### PR TITLE
feat(button): AR-2991 synchronize button component to design #patch

### PR DESCRIPTION
**Pull request by aziz-zen-ab**
This pull request synchronize the button component styling with the figma design: https://www.figma.com/file/LRwRrrykHF9SrTxBPuvp2c/%5BDRAFT%5D-AP---Master-Components-2.0?node-id=9%3A132. Several state styling need to change for example the hover button background color, color code for the warning appearance, etc.

This pull request also update the color variable library base on the color palette provided in the Figma design: https://www.figma.com/file/LRwRrrykHF9SrTxBPuvp2c/%5BDRAFT%5D-AP---Master-Components-2.0?node-id=325%3A3238

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "justice-ui-library",
-  "version": "0.4.0",
+  "version": "0.4.1",
```
Comment (aris-fm) [inline package.json:3]: I think if this is a major release, the version should become `1.0.0` instead

Comment (aziz-zen-ab) [reply package.json:3]: alright

**Comment by rayhan-ab** [comment]
you don't update the version? then why there's #minor in commit message? anyway, if it's only refactoring styles, then #patch is enough i think

**Comment by aziz-zen-ab** [comment]
> you don't update the version? then why there's #minor in commit message? anyway, if it's only refactoring styles, then #patch is enough i think

The Change log file has the 4.1 version as it latest but the package.json is not yet update, so I update it in this commit.

**Comment by rayhan-ab** [comment]
then please update the commit message to #patch and i suggest you to delete the tag as well so that this commit will be included in 0.4.1

**Comment by rayhan-ab** [comment]
if you don't update the version in package.json and include your changelog in 0.4.1 then no need to add # at the end of the commit message

File: src/components/styled-atlaskit/Button/Button.scss [hunk 2]
```diff
@@ -17,6 +17,9 @@ button.styled-atlaskit-button {
     font-size: $font-size-h6;
     line-height: 30px;
     font-weight: normal;
+    span {
```
Comment (rayhan-ab) [inline src/components/styled-atlaskit/Button/Button.scss:20]: please add > so that it will only style the direct child

**Comment by rayhan-ab** [comment]
i'm still kinda confused with the versioning here. if the commit message says #patch then why version in package.json is not updated? and why the change is placed in 0.4.1 in changelog?

**Comment by aziz-zen-ab** [comment]
> i'm still kinda confused with the versioning here. if the commit message says #patch then why version in package.json is not updated? and why the change is placed in 0.4.1 in changelog?

sorry not really aware that the current version is already 0.4.2

File: src/components/styled-atlaskit/Button/Button.scss (hunk 2 above)
Comment (rayhan-ab) [inline src/components/styled-atlaskit/Button/Button.scss:20]: can we take this out of button span selector? and instead declare it in new selector button > span > span?

**Comment by rayhan-ab** [comment]
still confused with your versioning intention, why now it changes to #minor but version in package.json is not changed?

File: src/components/styled-atlaskit/Button/Button.scss [hunk 3]
```diff
@@ -76,29 +80,138 @@ button.styled-atlaskit-button {
...
+      &:active, &.selected {
         background-color: $button-backgroundColor--subtle--active;
+        span {
+          &[class^="Spinner"] svg {
+            stroke: $base-white;
+          }
+        }
       }
-      &.danger {
+      &-link {
```
Comment (rayhan-ab) [inline src/components/styled-atlaskit/Button/Button.scss:91]: when does this selector happen?

Comment (aziz-zen-ab) [reply src/components/styled-atlaskit/Button/Button.scss:91]: this selector intended to handle the `appearance="subtle-link"` styling

**Comment by aziz-zen-ab** [comment]
> still confused with your versioning intention, why now it changes to #minor but version in package.json is not changed?

After I reread semantic versioning again. I realize it should be `patches` since it fixes the button styling, and adding colors. Sorry for the lack of versioning understanding before.

**Review by rayhan-ab — APPROVED** [review]

**Pull request merged**